pkg/form3/form3.go

```diff
@@ -2,81 +2,95 @@ package form3
 
 import (
 	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
 	"net/http"
 	"net/url"
 	"os"
-	"strings"
 
-	"github.com/form3tech-oss/go-form3/pkg/generated/client"
+	genClient "github.com/form3tech-oss/go-form3/pkg/generated/client"
 	rc "github.com/go-openapi/runtime/client"
 	"github.com/go-openapi/strfmt"
 )
 
+const (
+	ReqMimeType = "application/vnd.api+json"
+)
+
 type F3 struct {
-	client.Form3Public
-	Defaults *TestDefaults
+	genClient.Form3Public
+	Defaults *ClientDefaults
 }
 
-type Config struct {
-	Host           string
-	ClientId       string
-	ClientSecret   string
-	OrganisationId string
-	HttpDebug      bool
-	schemes        []string
+func New(host, pubKeyID string, privateKey *rsa.PrivateKey, orgID string) *F3 {
+	u, err := url.Parse(host)
+	if err != nil {
+		panic(err)
+	}
+
+	config := NewRequestSigningClientConfig(pubKeyID, privateKey)
+	httpClient := NewRequestSigningHTTPClient(config)
+
+	return newF3(u, httpClient, orgID)
 }
 
-func NewConfig(host, clientID, secret, organisationID string) Config {
-	var schemes []string
-	if strings.HasPrefix(host, "https://") {
-		schemes = append(schemes, "https")
-	} else if strings.HasPrefix(host, "http://") {
-		schemes = append(schemes, "http")
-	} else {
-		panic("Unsupported host protocol")
+func NewFromEnv() *F3 {
```

Comment:
It would be nice if we had a `NewFromEnv` func that returned a form3 client either with request signing or client id and secret setup inside based on the environment variables you have setup.  Rather than having to set the env variables then call a different func based on the client you want.

Reply:
The rationale to not doing it like that was that the token-based setup was to be deprecated, so the "default" functions would only contain code for using request signing. I don't mind merging the code together though so that the functions will return request signing client and fallback to token-based automatically.

Reply:
ok fair enough